src/components/DateInput.jsx

```diff
@@ -81,13 +87,29 @@ export default class DateInput extends React.Component {
   }
 
   onKeyDown(e) {
```

Comment:
Debouncing doesn't actually make very much sense here, because the `onKeyDown` listener will still be called for every keystroke, it'll just be called on a delay and there's no way to ostensibly cancel the previous keydown listener. Maybe `throttle is more reasonable?